data-pipeline/tests/test_trace_exporter.rs

```diff
@@ -30,6 +33,37 @@ mod tracing_integration_tests {
         tinybytes::Bytes::from(encoded_data)
     }
 
+    fn get_v05_trace_snapshot_test_payload() -> Bytes {
+        let mut dict = SharedDict::default();
+
+        let span_1 = create_test_v05_span(
+            1234,
+            12342,
+            12341,
+            1,
+            false,
+            &mut dict,
+            Some(vec![
+                ("_dd_metric1".to_string(), 1.1),
+                ("_dd_metric2".to_string(), 2.2),
+            ]),
+        );
+        let span_2 = create_test_v05_span(1234, 12343, 12341, 1, false, &mut dict, None);
```

Comment:
For v0.4 tests we serialize json rather than using the same Span struct we deserialize with to avoid masking potential flaws in our model. Can / should we do the same for v0.5?

Reply:
Oh, with v05 is a problem because there are some incompatibilities between the formats. The one affecting us is the fact that json can't have numbers as keys in fields so meta and metrics can't be represented directly from json unless we implement some custom serialization for that json objects.
One thing that we could do is maybe generate the payloads from a reference implementation and use that both for data-pipeline and the agent.